charts/operator-wandb/charts/app/templates/_deployment.tpl

```diff
@@ -0,0 +1,291 @@
+{{/*
+This template is used to generate the deployment for the app, and is used for both the non-glue and glue deployments.
+*/}}
+{{- define "app.deployment" -}}
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ include "app.fullname" . }}{{ .suffix }}
+  labels:
+    {{- include "wandb.commonLabels" . | nindent 4 }}
+    {{- include "app.commonLabels" . | nindent 4 }}
+    {{- include "app.labels" . | nindent 4 }}
+    {{- if .Values.deployment.labels }}
+    {{-   toYaml .Values.deployment.labels | nindent 4 }}
+    {{- end }}
+  annotations:
+    {{- include "wandb.deploymentAnnotations" . | nindent 4 }}
+    {{- if .Values.deployment.annotations }}
+    {{-   toYaml .Values.deployment.annotations | nindent 4 }}
+    {{- end }}
+spec:
+  replicas: 1
+  {{- if and .Values.hpa.enabled .Values.glueSingleton.enabled }}
+  strategy:
+    type: RollingUpdate
+    rollingUpdate:
+      maxUnavailable: 0
+      maxSurge: 1
+  {{- end }}
```

Comment:
Only setting the rollout strategt if both hpa and glueSingleton enabled.

Reply:
Why do we need to do all this? Why cannot we just write it out?

Reply:
I would perfer to be explict then implicit. This adds some magic, that we plan on decoupling anyway

Reply:
This was added to make sure that when an upgrade to the spec happens, it actually rolls out properly one by one. 

Otherwise it will kill all the pods but one and replace them in bulk.